drivers/gpio/gpio_numicro.c

```diff
@@ -162,10 +166,23 @@ static int gpio_numicro_pin_interrupt_configure(const struct device *dev,
 					     enum gpio_int_trig trig)
 {
 	const struct gpio_numicro_config *cfg = dev->config;
+#ifdef CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT
+	struct gpio_numicro_data *data = dev->data;
+#endif /* CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT */
 	uint32_t int_type = 0;
 	uint32_t int_level = 0;
 	uint32_t int_level_mask = BIT(pin) | BIT(pin + 16);
 
+#ifdef CONFIG_GPIO_ENABLE_DISABLE_INTERRUPT
+	if (mode == GPIO_INT_MODE_DISABLE_ONLY) {
+		cfg->regs->INTEN &= ~(BIT(pin) | BIT(pin + 16));
+		return 0;
```

Comment:
Note - you're not updating the `data->interrupt_en_reg` cached copy in this (or the INT_MODE_ENABLE_ONLY) path.

Reply:
This is the enable and disable only path, so in enabling branch, we restore the `INTEN` register by referencing what we recorded in `interrupt_en_reg`; in disabling branch, we simply disable the `INTEN` register. Thus, there is no need to update the `interrupt_en_reg` here. Maybe I should use a better naming for this register.